### PR TITLE
chore: remove unused AI and Daily OS code

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary

- remove unused AI function-result metadata, generated deserializers, prompt-card state, and their test-only callers
- remove unused Daily OS wake-context fields, learning-bullet deserialization, checklist correction metadata, category icon suggestion logic, and GetIt test shims
- document and suppress 51 Drift schema-column false positives at the declarative table source because those getters are consumed during code generation
- update test-review notes so they describe the remaining production API

## DCM impact

The authoritative baseline was 171. Merged PR #3779 removed 8 genuine declarations, so the expected starting point for this PR is 163.

This PR removes 15 additional genuine unused declarations and suppresses 51 generator-required Drift getter reports. The expected result is therefore approximately 97 unused-code reports. DCM is not installed in the Linux development VM, so the licensed macOS command remains the authoritative post-merge verification.

## Validation

- `make analyze` — no issues
- 16 targeted test files — 493 tests passed
- `git diff --check` — clean

The roughly 27,000-test full suite is intentionally delegated to CI, where it runs in 10 shards. This is internal cleanup with no user-visible UI change, so no screenshot or changelog entry is needed.

## Summary by CodeRabbit

* **Refactor**
  * Simplified AI function results to retain only outcome data, status, and errors.
  * Streamlined AI input and planner context models by removing unused metadata and deserialization paths.
  * Removed legacy category-name icon suggestions and redundant correction metadata.
  * Simplified generated prompt and learning-model APIs.

* **Tests**
  * Updated test coverage and fixtures to match the streamlined APIs.
  * Removed obsolete tests for retired functionality while preserving relevant serialization and behavior coverage.

* **Documentation**
  * Updated test-review documentation to reflect completed cleanup and remaining coverage items.

### CODE REVIEW

File: test/features/categories/TEST_REVIEW.md [hunk 1]
```diff
@@ -134,9 +129,9 @@
 
 ## Summary
 
-- **2 oversized impl files**: `category_icon.dart` (950 lines — clear 3-way split); `category_details_page.dart` (753 lines — widget extraction).
+- **1 oversized impl file**: `category_details_page.dart` (753 lines — widget extraction). `category_icon.dart` is down to 306 lines after table extraction and unused-code removal.
 - **2 severely oversized test files**: `category_details_page_test.dart` (1955 lines, 86 `pumpAndSettle`) and `category_details_controller_test.dart` (1350 lines, 21 real-time timeouts) — both need `_pumpPage`/`_TestBench` helpers and file-level splits.
 - **5 weak/smoke tests**: `controller state updates trigger UI rebuilds` (asserts only `verify().called(1)`); `dispose does not throw` (asserts only text render); `CategoryIconConstants` size checks; `handles stream errors` (asserts only `findsOneWidget`); Performance map test (re-tests `fromJson` already covered).
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/categories/TEST_REVIEW.md]: Updated the category review throughout: current file and test sizes, resolved helper status, removed smoke-test count, current serialization lines, and current missing-test inventory.

File: test/features/daily_os_next/agents/domain/TEST_REVIEW.md [hunk 2]
```diff
@@ -13,10 +13,10 @@ Generated by automated test-quality review (static analysis).
 |------|-------|-----------|-----------|
 | `lib/.../domain/day_agent_config.dart` | 35 | No | No test file at all; `toJson` round-trip untested |
 | `lib/.../domain/day_agent_kind.dart` | 16 | No | No test file; `DayAgentKind.matches` untested |
-| `lib/.../domain/day_agent_plan_models.dart` | 209 | Yes | `DayAgentLearningBullet`/`DayAgentLearningCard` lack Glados round-trip property |
+| `lib/.../domain/day_agent_plan_models.dart` | 187 | Yes | Learning models expose serialization only; no unused deserialization API remains |
 | `lib/.../domain/day_agent_reconcile_models.dart` | 373 | Yes | No issues (excellent Glados coverage) |
 | `lib/.../domain/day_agent_slots.dart` | 21 | Yes | `dayAgentPlanEntityId` is not exercised in the test |
-| `test/.../domain/day_agent_plan_models_test.dart` | 173 | — | Missing Glados property for `DayAgentLearningBullet`/`Card` |
+| `test/.../domain/day_agent_plan_models_test.dart` | 198 | — | Learning-card serialization and equality are covered |
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/daily_os_next/agents/domain/TEST_REVIEW.md:19]: Updated both inventory rows to state that dayAgentPlanEntityId has static and Glados coverage.

File: test/TEST_REVIEW.root.md [hunk 3]
```diff
@@ -63,10 +63,8 @@ Generated by automated test-quality review (static analysis).
...
-  **RESOLVED:** added `'delegates to logging service on error messages (success path)'` — stubs `error` to succeed, asserts `error(LogDomain.settings, msg, subDomain: 'error')` is called once and the DevLogger fallback stays empty.
+- [x] **[MED]** `lib/get_it.dart:172` — `registerSingletons()` (the main production bootstrap function) has **no direct unit-test coverage**. Testing it fully requires integration infrastructure because it instantiates roughly 30 database and platform services. **ASSESSED:** the former public testing shim duplicated private bootstrap helpers without production callers and was removed; app startup and integration tests remain the appropriate coverage level for this orchestration.
+- [x] **[LOW]** The former `safeLogForTesting` shim duplicated a private logging helper solely to expose it to tests. **RESOLVED:** the unused shim and its implementation-coupled tests were removed.
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/TEST_REVIEW.root.md:69]: Removed the stale critical Glados summary and synchronized the root review table and resolved VectorClock merge coverage with the current tests.

File: test/TEST_REVIEW.root.md [hunk 4]
```diff
@@ -15,11 +15,11 @@ Generated by automated test-quality review (static analysis).
 
 | File | Lines | Has test? | Top issue |
 |------|-------|-----------|-----------|
-| `lib/features/sync/vector_clock.dart` | 137 | Yes (both static and Glados) | `merge` and `mergeUniqueClocks` methods untested |
-| `test/vector_clock_glados_test.dart` | 111 | n/a | Missing `tags: 'glados'` on all four `Glados2` calls |
-| `test/vector_clock_test.dart` | 63 | n/a | `merge` / `mergeUniqueClocks` / `fromJson` / `toJson` coverage gaps |
-| `lib/get_it.dart` | 856 | Partial | 856-line file; only 3 functions tested; `registerSingletons` orchestration completely untested |
-| `test/get_it_test.dart` | 455 | n/a | Mock setup is proportionate; test quality is good; missing scope isolation edge cases |
+| `lib/features/sync/vector_clock.dart` | 154 | Yes (both static and Glados) | Merge and serialization contracts are covered |
+| `test/vector_clock_glados_test.dart` | 191 | n/a | Glados tags and algebraic merge properties are present |
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/TEST_REVIEW.root.md:19]: Fixed in follow-up PR #3782 at 4dad9cc90: the Files reviewed inventory now uses the same current VectorClock and test sizes as the table.

File: test/features/categories/TEST_REVIEW.md [hunk 5]
```diff
@@ -134,9 +129,8 @@
...
-- **2 prime Glados candidates**: `suggestFromName` (complex 3-stage matching with threshold arithmetic) and `fromJson`/`toJson` roundtrip on arbitrary strings — both are pure functions on structured input.
-- **3 missing test files**: `categories_list_controller.dart`, `category_color_icon.dart`, `category_icon_compact.dart` (+ `category_type_card.dart`).
+- **1 oversized impl file**: `category_details_page.dart` is now 521 lines after form-section extraction. `category_icon.dart` is down to 306 lines after table extraction and unused-code removal.
+- **2 long test files**: `category_details_page_test.dart` is 1718 lines and `category_details_controller_test.dart` is 1108 lines. Their shared pump/container helpers are implemented; further file splitting is constrained by the one-test-file-per-source rule.
+- **1 prime Glados candidate**: `fromJson`/`toJson` roundtrip on arbitrary strings, a pure function on structured input.
+- **2 missing test files**: `category_color_icon.dart` and `category_type_card.dart`. `categories_list_controller.dart` and `category_icon_compact.dart` now have mirror tests.
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/categories/TEST_REVIEW.md:135]: Fixed in follow-up PR #3782 at 4dad9cc90: the per-file table now records the existing category list and compact icon mirror tests and their current coverage.

File: test/features/daily_os_next/agents/domain/TEST_REVIEW.md [hunk 6]
```diff
@@ -13,12 +13,12 @@ Generated by automated test-quality review (static analysis).
...
-| `lib/.../domain/day_agent_plan_models.dart` | 209 | Yes | `DayAgentLearningBullet`/`DayAgentLearningCard` lack Glados round-trip property |
+| `lib/.../domain/day_agent_plan_models.dart` | 187 | Yes | Learning models expose serialization only; no unused deserialization API remains |
 | `lib/.../domain/day_agent_reconcile_models.dart` | 373 | Yes | No issues (excellent Glados coverage) |
-| `lib/.../domain/day_agent_slots.dart` | 21 | Yes | `dayAgentPlanEntityId` is not exercised in the test |
-| `test/.../domain/day_agent_plan_models_test.dart` | 173 | — | Missing Glados property for `DayAgentLearningBullet`/`Card` |
+| `lib/.../domain/day_agent_slots.dart` | 21 | Yes | `dayAgentPlanEntityId` has static and Glados coverage |
+| `test/.../domain/day_agent_plan_models_test.dart` | 198 | — | Learning-card serialization and equality are covered |
 | `test/.../domain/day_agent_reconcile_models_test.dart` | 668 | — | None; gold-standard example |
-| `test/.../domain/day_agent_slots_test.dart` | 102 | — | `dayAgentPlanEntityId` helper is never called |
+| `test/.../domain/day_agent_slots_test.dart` | 102 | — | Covers `dayAgentPlanEntityId` statically and with Glados |
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/daily_os_next/agents/domain/TEST_REVIEW.md:21]: Fixed in follow-up PR #3782 at 4dad9cc90: the Daily OS inventory now uses the current 324-line slot test and synchronizes the full domain inventory.
